### PR TITLE
feat(sheet): add component tokens

### PR DESCRIPTION
**Pull request by alisonailea**
**Related Issue:** #7180

### Sheet

--calcite-sheet-scrim-background: [Deprecated] use `--calcite-sheet-scrim-background-color` instead. Specifies the background color of the sheet scrim.
--calcite-sheet-scrim-background-color: Specifies the background color of the sheet scrim.
--calcite-sheet-background-color: Specifies the background color of the sheet.
--calcite-sheet-shadow: Specifies the shadow of the sheet.
--calcite-sheet-text-color: Specifies the text color of the sheet.

### CODE REVIEW

File: packages/calcite-components/src/components/sheet/sheet.scss [hunk 1]
```diff
@@ -3,7 +3,11 @@
  *
  * These properties can be overridden using the component's tag as selector.
  *
- * @prop --calcite-sheet-scrim-background: Specifies the background color of the sheet scrim.
+ * @prop --calcite-sheet-scrim-background: [Deprecated] use `--calcite-sheet-scrim-background-color` instead. Specifies the background color of the sheet scrim.
+ * @prop --calcite-sheet-scrim-background-color: Specifies the background color of the sheet scrim.
+ * @prop --calcite-sheet-background-color: Specifies the background color of the sheet.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/sheet/sheet.scss:8]: I wonder if this needs to have a BG set at all. I'd probably vote to just remove this and have the container display with transparent BG. A user should be slotting a Panel, etc., so this doesn't seem needed.

File: packages/calcite-components/src/components/sheet/sheet.scss [hunk 2]
```diff
@@ -3,7 +3,11 @@
...
+ * @prop --calcite-sheet-shadow: Specifies the shadow of the sheet.
+ * @prop --calcite-sheet-text-color: Specifies the text color of the sheet.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/sheet/sheet.scss:10]: I don't think we need text-color, everything is slotted by a user.

File: packages/calcite-components/src/components/sheet/sheet.scss (hunk 1 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/sheet/sheet.scss:8]: This would apply to modal too.

Comment (macandcheese) [reply packages/calcite-components/src/components/sheet/sheet.scss:8]: Well, Modal does currently have bg color (and separate for Content area). It’s a bit different than Sheet as you don’t slot a Panel into a Modal. That is the current behavior anyway, we have discussed changing that to align more with Sheet, Shell panel, etc.

Comment (jcfranco) [reply packages/calcite-components/src/components/sheet/sheet.scss:8]: Gotcha. Do we have an issue for that? For future reference purposes.

Comment (alisonailea) [reply packages/calcite-components/src/components/sheet/sheet.scss:8]: would that apply to shell panel too?

Comment (macandcheese) [reply packages/calcite-components/src/components/sheet/sheet.scss:8]: #7886 would address that I think.

**Review by jcfranco — APPROVED** [review]
📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄
📄✅📄📄📄✅✅✅✅📄✅✅✅✅✅📄✅📄📄📄✅📄✅📄
📄✅📄📄📄✅📄📄📄📄📄📄✅📄📄📄✅✅📄✅✅📄✅📄
📄✅📄📄📄✅📄✅✅📄📄📄✅📄📄📄✅📄✅📄✅📄✅📄
📄✅📄📄📄✅📄📄✅📄📄📄✅📄📄📄✅📄📄📄✅📄📄📄
📄✅✅✅📄✅✅✅✅📄📄📄✅📄📄📄✅📄📄📄✅📄✅📄
📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄📄

**Pull request merged**